### ISSUE TITLE
Sync username restrictions with gin-repo

### ISSUE THREAD

**Issue by gicmo** [G-Node/gin-auth#105]
Currently gin-repo requires usernames to be at least 3 characters long and start with a letter or a number. gin-auth currently has a different set of rules. They need to be in sync.

**Comment by mpsonntag** [comment]
Since we also use a set of rules for username at gin-ui its probably best to manage these rules via gin-core and import them into all projects.

**Comment by achilleas-k** [comment]
Agreed. We need to start using gin-core for all common functionality.
